book/analysis/big-o-notation.md

```diff
@@ -5,104 +5,106 @@ position: 2
+an algorithm performs and determine the dominant part of the $$T(n)$$ function.
+We can do this because, as the problem gets larger, some portion of the $$T(n)$$
+function tends to overpower the rest; it is this dominant portion that is
+ultimately most helpful for algorithm comparisons.
+
+The *order of magnitude* function describes the part of $$T(n)$$ that increases
+fastest as the value of *n* increases. “Order of magnitude function” is a bit of
+a mouthful, though, so we call it *big O* notation and write it as $$O(f(n))$$,
+where $$f(n)$$ is the dominant part of the original $$T(n)$$. Big O notation provides a useful approximation for the *actual* number of steps in a
+computation.
+
+In the above example, we saw that $$T(n)=1+n$$. As *n* gets larger, the constant
+1 will become less significant to the final result. If we are simply looking for
+an approximation of $$T(n)$$, then we can drop the 1 and say that the
+running time is $$O(n)$$.
+
+Let's be clear, though: the 1 *is* important to $$T(n)$$ and can only be safely
+ignored when we are looking for an *approximation* of $$T(n)$$.
 
 As another example, suppose that for some algorithm, the exact number of
-steps is $$T(n)=5n^{2}+27n+1005$$. When *n* is small, say 1 or 2, the
-constant 1005 seems to be the dominant part of the function. However, as
-*n* gets larger, the $$n^{2}$$ term becomes the most important. In fact,
-when *n* is really large, the other two terms become insignificant in
-the role that they play in determining the final result. Again, to
-approximate $$T(n)$$ as *n* gets large, we can ignore the other terms and
-focus on $$5n^{2}$$. In addition, the coefficient $$5$$ becomes
-insignificant as *n* gets large. We would say then that the function
-$$T(n)$$ has an order of magnitude $$f(n)=n^{2}$$, or simply that it is
-$$O(n^{2})$$.
-
-Although we do not see this in the summation example, sometimes the
-performance of an algorithm depends on the exact values of the data
-rather than simply the size of the problem. For these kinds of
-algorithms we need to characterize their performance in terms of best
-case, *worst case*, or *average case* performance. The worst case
-performance refers to a particular data set where the algorithm performs
-especially poorly. Whereas a different data set for the exact same
-algorithm might have extraordinarily good performance. However, in most
-cases the algorithm performs somewhere in between these two extremes
-(average case). It is important for a computer scientist to understand
-these distinctions so they are not misled by one particular case.
-
-A number of very common order of magnitude functions will come up over
-and over as you study algorithms. These are shown in
-the table below. In order to decide which of these
-functions is the dominant part of any $$T(n)$$ function, we must see how
-they compare with one another as *n* gets large.
+steps is $$T(n)=5n^{2}+27n+1005$$. When *n* is small (1 or 2), the constant 1005
```

Comment:
@aleksharma12 I know you just carried this over from the prior version, but I think it would be best to render n in the same style as T(n), so we should wrap it in $$dollar signs$$ to have it parsed by katex.